modules/Microsoft.Compute/virtualMachines/.bicep/nested_networkInterface.bicep

```diff
@@ -65,7 +65,7 @@ module networkInterface '../../../Microsoft.Network/networkInterfaces/deploy.bic
       name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
       primary: index == 0
       privateIPAllocationMethod: contains(ipConfiguration, 'privateIPAllocationMethod') ? (!empty(ipConfiguration.privateIPAllocationMethod) ? ipConfiguration.privateIPAllocationMethod : null) : null
-      privateIPAddress: contains(ipConfiguration, 'vmIPAddress') ? (!empty(ipConfiguration.vmIPAddress) ? ipConfiguration.vmIPAddress : null) : null
+      vmIPAddress: contains(ipConfiguration, 'vmIPAddress') ? (!empty(ipConfiguration.vmIPAddress) ? ipConfiguration.vmIPAddress : null) : null
```

Comment:
Hey @chris5287, I would actually argue the opposite: We should update the NIC module as it's interface is incorrect.

Reviewing all properties I'd argue:
```Bicep
ipConfigurations: [for (ipConfiguration, index) in ipConfigurations: {
  name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
  properties: {
    primary: index == 0 ? true : false
    privateIPAllocationMethod: contains(ipConfiguration, 'privateIPAllocationMethod') ? (!empty(ipConfiguration.privateIPAllocationMethod) ? ipConfiguration.privateIPAllocationMethod : null) : null
    privateIPAddress: contains(ipConfiguration, 'vmIPAddress') ? (!empty(ipConfiguration.vmIPAddress) ? ipConfiguration.vmIPAddress : null) : null
    publicIPAddress: contains(ipConfiguration, 'publicIPAddressResourceId') ? (ipConfiguration.publicIPAddressResourceId != null ? {
      id: ipConfiguration.publicIPAddressResourceId
    } : null) : null
    subnet: {
      id: ipConfiguration.subnetId
    }
    loadBalancerBackendAddressPools: contains(ipConfiguration, 'loadBalancerBackendAddressPools') ? ipConfiguration.loadBalancerBackendAddressPools : null
    applicationSecurityGroups: contains(ipConfiguration, 'applicationSecurityGroups') ? ipConfiguration.applicationSecurityGroups : null
    applicationGatewayBackendAddressPools: contains(ipConfiguration, 'applicationGatewayBackendAddressPools') ? ipConfiguration.applicationGatewayBackendAddressPools : null
    gatewayLoadBalancer: contains(ipConfiguration, 'gatewayLoadBalancer') ? ipConfiguration.gatewayLoadBalancer : null
    loadBalancerInboundNatRules: contains(ipConfiguration, 'loadBalancerInboundNatRules') ? ipConfiguration.loadBalancerInboundNatRules : null
    privateIPAddressVersion: contains(ipConfiguration, 'privateIPAddressVersion') ? ipConfiguration.privateIPAddressVersion : null
    virtualNetworkTaps: contains(ipConfiguration, 'virtualNetworkTaps') ? ipConfiguration.virtualNetworkTaps : null
  }
}]
```

should be changed to

```Bicep
ipConfigurations: [for (ipConfiguration, index) in ipConfigurations: {
  name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
  properties: {
    primary: index == 0 ? true : false
    privateIPAllocationMethod: contains(ipConfiguration, 'privateIPAllocationMethod') ? (!empty(ipConfiguration.privateIPAllocationMethod) ? ipConfiguration.privateIPAllocationMethod : null) : null
    privateIPAddress: contains(ipConfiguration, 'privateIPAddress') ? (!empty(ipConfiguration.privateIPAddress) ? ipConfiguration.privateIPAddress : null) : null
    publicIPAddress: contains(ipConfiguration, 'publicIPAddressResourceId') ? (ipConfiguration.publicIPAddressResourceId != null ? {
      id: ipConfiguration.publicIPAddressResourceId
    } : null) : null
    subnet: {
      id: ipConfiguration.subnetResourceId
    }
    loadBalancerBackendAddressPools: contains(ipConfiguration, 'loadBalancerBackendAddressPools') ? ipConfiguration.loadBalancerBackendAddressPools : null
    applicationSecurityGroups: contains(ipConfiguration, 'applicationSecurityGroups') ? ipConfiguration.applicationSecurityGroups : null
    applicationGatewayBackendAddressPools: contains(ipConfiguration, 'applicationGatewayBackendAddressPools') ? ipConfiguration.applicationGatewayBackendAddressPools : null
    gatewayLoadBalancer: contains(ipConfiguration, 'gatewayLoadBalancer') ? ipConfiguration.gatewayLoadBalancer : null
    loadBalancerInboundNatRules: contains(ipConfiguration, 'loadBalancerInboundNatRules') ? ipConfiguration.loadBalancerInboundNatRules : null
    privateIPAddressVersion: contains(ipConfiguration, 'privateIPAddressVersion') ? ipConfiguration.privateIPAddressVersion : null
    virtualNetworkTaps: contains(ipConfiguration, 'virtualNetworkTaps') ? ipConfiguration.virtualNetworkTaps : null
  }
}]
``` 

effectively updating both the `privateIPAddress` & `subnet` property.
```

Reply:
Naturally that would mean we'd need to update also the NIC module's test file & the VM modules handover (as the `SubnetId` would now be `SubnetResourceId`). 

Would you agree?

Reply:
Ah cool, I wasn't sure how you would feel about this as it's a breaking change for the current CARML interface.

Reply:
It sure is - but as we don't yet have a strict breaking change policy (pre V1.0) I'd argue now is the best time to make it 'right'.